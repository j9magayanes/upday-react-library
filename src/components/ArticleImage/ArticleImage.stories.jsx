import React from 'react';

import { ArticleImage } from './ArticleImage';


export default {
  title: 'Form/ArticleImage',
  component:ArticleImage,
};

const Template = (args) =><ArticleImage {...args} />;



export const Web = Template.bind({});
Web.args = {
    src: "https://img-staging.yana.asideas.de/image/fetch/w_700,h_394,c_fill,f_webp,q_auto:eco/https%3A%2F%2Fcdn-images.rtp.pt%2Fnoticias%2Fimages%2Fdefault_noticias.png%3Fw%3D860%26q%3D90%26auto%3Dformat",
    alt: "test"

};