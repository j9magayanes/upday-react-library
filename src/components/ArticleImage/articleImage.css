.hero {

    min-width: 100% !important;
    max-width: 100% !important;
    min-height:  100% !important;
    max-height:  100% !important;
}