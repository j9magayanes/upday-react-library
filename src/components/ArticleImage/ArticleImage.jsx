import React from 'react';
import  'upday-web-components/components/articleImage'

export const ArticleImage = ({ src, alt, imageRef, photoCaption, figureRef}) => {
    return (
        <article-image src={src} alt={alt} imageRef={imageRef ? imageRef: null } photoCaption={photoCaption} figureRef={figureRef}/>
    )
}