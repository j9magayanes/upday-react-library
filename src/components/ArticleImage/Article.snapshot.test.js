import React from 'react';
import renderer from 'react-test-renderer';
import { ArticleImage } from './ArticleImage';

test('renders ArticleImage component correctly', () => {
  const props = {
    src: 'your-image-src',
    alt: 'your-alt-text',
    imageRef: null, // or provide a valid ref if needed
    photoCaption: 'your-photo-caption',
    figureRef: null, // or provide a valid ref if needed
  };

  const component = renderer.create(<ArticleImage {...props} />);
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
