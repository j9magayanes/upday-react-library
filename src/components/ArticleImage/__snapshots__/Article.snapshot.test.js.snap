// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ArticleImage component correctly 1`] = `
<article-image
  alt="your-alt-text"
  figureRef={null}
  imageRef={null}
  photoCaption="your-photo-caption"
  src="your-image-src"
/>
`;
